src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java

```diff
@@ -0,0 +1,304 @@
+package bio.terra.service.snapshotbuilder.utils;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.equalToCompressingWhiteSpace;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.common.category.Unit;
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.grammar.azure.SynapseVisitor;
+import bio.terra.model.SnapshotBuilderCriteria;
+import bio.terra.model.SnapshotBuilderCriteriaGroup;
+import bio.terra.model.SnapshotBuilderDomainCriteria;
+import bio.terra.model.SnapshotBuilderProgramDataListCriteria;
+import bio.terra.model.SnapshotBuilderProgramDataRangeCriteria;
+import bio.terra.service.snapshotbuilder.query.FilterVariable;
+import bio.terra.service.snapshotbuilder.query.Query;
+import bio.terra.service.snapshotbuilder.query.TablePointer;
+import bio.terra.service.snapshotbuilder.query.TableVariable;
+import java.math.BigDecimal;
+import java.util.List;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@Tag(Unit.TAG)
+class CriteriaQueryBuilderTest {
+
+  @Test
+  void generateRangeCriteriaFilterProducesCorrectSql() {
+    SnapshotBuilderProgramDataRangeCriteria rangeCriteria = generateRangeCriteria();
+    FilterVariable filterVariable =
+        new CriteriaQueryBuilder("person", SynapseVisitor.azureTableName("source_dataset_name"))
```

Comment:
I don't think this test uses the `TableGenerator` argument so it can be `null`
```suggestion
        new CriteriaQueryBuilder("person", null)
```

or if you don't want to pass null, you could provide a no-op function
```suggestion
        new CriteriaQueryBuilder("person", s -> s)
```

In general I think all the tests here shouldn't use `SynapseVisitor` at all as it isn't testing that code, and if it needs a valid `TableGenerator` it can create one locally, e.g., as a lambda expression.